### PR TITLE
use shelljs for shell like commands

### PR DESCRIPTION
**Pull request by mikekidder**
This is preliminary;  unsure of the **test:cov** and **test:watch**..  The latter is causing a loop

### CODE REVIEW

**Comment by mikekidder** [comment]
Also note that ansi codes currently will not come through on output.

**Comment by gaearon** [comment]
@mikekidder Do you need help fixing those problems?

**Comment by mikekidder** [comment]
1) **test:cov** - may be a mocha issue w/windows, doesn't appear you are having error. Correct?
2) **test:watch** - may be related to **test:cov**, but the --watch may probablematic for shelljs
3) colors output - shelljs using **child-process** internally; known issue https://github.com/arturadib/shelljs/issues/86

If you want to get away from npm script, as you mention in #94, could also look into gulp.

**Comment by gaearon** [comment]
Woah, color support is a bummer! The issue has been up for two years..

Can you help me with understanding what exactly fails on Windows? How are shell scripts different from the same commands right inside `package.json`? Why do these commands work, but shell scripts don't work?

**Comment by mikekidder** [comment]
No difference in running npm script vs. shelljs.  The test scripts were put in after my initial PR's, and hadn't ran them yet until yesterday.

This fixes **test:cov** as mentioned in https://github.com/gotwarlost/istanbul/issues/90. Path to mocha should work on Mac as well.   And looks like **test:watch** is working.

This really leaves no color output as the issue.

**Comment by gaearon** [comment]
Thanks! I'll take another look later today.

File: scripts/prepublish.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+var sh = require('shelljs');
+
+sh.exec('eslint src test', function(code, output) {
+
+  if(code === 0){
```
Comment (gaearon) [inline scripts/prepublish.js]: Nit pick: missing space before `{` and unneeded newlines before and after `if` block.

File: scripts/prepublish.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+var sh = require('shelljs');
+
+sh.exec('eslint src test', function(code, output) {
+
```
Comment (gaearon) [inline scripts/prepublish.js]: Let's also run the tests here.

**Comment by gaearon** [comment]
This PR should also remove the old scripts, right?

**Comment by gaearon** [comment]
Sorry for dragging my feet here. I'm very busy atm, but will be more available in a week or so.
I'll definitely either merge this or go back to npm scripts.

Just give me some more time before looking at it!
Cheers

**Comment by mikekidder** [comment]
Oh, no worries from me. I know how to work around the shell scripts. :)  
FWIW, the npm scripts worked and has color output.

**Comment by gaearon** [comment]
Closing in favor of #267.   
Nevertheless thanks for working on this!

**Comment by gaearon** [comment]
Fixed by #267, the fix is merged into `breaking-changes-1.0` branch. Feel free to give it a go!

**Pull request closed without merging**